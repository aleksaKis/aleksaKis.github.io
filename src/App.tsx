import { useScroll } from 'framer-motion';
import { useRef } from 'react';

import { Experience } from './components/containers/experience';
import { Landing } from './components/containers/landing';
import { Showcase } from './components/containers/showcase';
import { AnimatedAuroras } from './components/ui/animated-auroras';
import { SparklesCore } from './components/ui/sarkles';

function App() {
    const mainRef = useRef<HTMLDivElement>(null);

    const { scrollYProgress } = useScroll({ container: mainRef });
    return (
        <main className="overflow-x-hidden relative">
            <div
                ref={mainRef}
                className="bg-stone-950 text-neutral-300 h-screen w-screen overflow-x-hidden overflow-y-auto"
            >
                <div className="top-0 w-full absolute inset-0 h-screen pointer-events-none">
                    <SparklesCore
                        background="transparent"
                        minSize={0.3}
                        maxSize={2}
                        particleDensity={10}
                        className="w-full h-full"
                        particleColor="#FFFFFF"
                    />
                </div>
                <AnimatedAuroras scrollYProgress={scrollYProgress} />
                <div className="max-w-[1280px] mx-auto px-6">
                    <Landing />
                    <Experience />
                    <Showcase />
                </div>
            </div>
        </main>
    );
}

export default App;
