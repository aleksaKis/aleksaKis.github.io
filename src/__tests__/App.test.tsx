import { render, screen } from '@testing-library/react';

import App from '../App';

describe('<App/>', () => {
    it('should match the snapshot', () => {
        const { container } = render(<App />);
        expect(container).toMatchSnapshot();
    });

    it('should render the text "Hello world!"', () => {
        render(<App />);
        expect(screen.getByText('Hello world!')).toBeInTheDocument();
    });
});
