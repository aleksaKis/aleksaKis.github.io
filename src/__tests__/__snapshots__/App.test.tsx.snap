// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App/> should match the snapshot 1`] = `
<div>
  <h1
    class="text-3xl font-bold underline"
  >
    Hello world!
  </h1>
</div>
`;
